postgres_composite_types/__init__.py

```diff
@@ -258,10 +259,32 @@ def __init__(cls, name, bases, attrs):
         if name == 'CompositeType':
             return
 
+        cls._capture_descriptors()
+
         # Register the type on the first database connection
         connection_created.connect(receiver=cls.database_connected,
                                    dispatch_uid=cls._meta.db_type)
 
+    def _capture_descriptors(cls):
+        """ Work around for not being able to call contribute_to_class."""
```

Comment:
I'm not sure what you mean. I can't think of any way to avoid calling `type(model_name, (models.Model,), attrs)`, as the Django model metaclass `__new__` method is what sets up `_meta` needed to call `contribute_to_class` on the model fields. Some of Django's built in field classes rely on even more Django internals, e.g. take a look at `DateField.contribute_to_class` (that's the one that made me give up on being able to avoid another call to `type`), but even if we faked that, there's no guarantee that some other third party package doesn't use a different part of django Model API which won't be present on `CompositeType`. 

Maybe I'm missing something though?